homeassistant/components/sql/__init__.py

```diff
@@ -45,6 +45,17 @@ def validate_sql_select(value: str) -> str:
 )
 
 
+def remove_configured_db_url_if_not_needed(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> None:
+    """Remove db url from config if it matches recorder database."""
+    new_options = {**entry.options, **{CONF_DB_URL: None}}
```

Comment:
I think it would be best to drop the key out of the dictionary

Reply:
That would probably make more sense. I'll change that.